app/knight.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+
+class Knight:
+    def __init__(self, knight_information: dict) -> None:
+        self.knight_information = knight_information
+        self.name = knight_information.get("name")
+        self.power = knight_information.get("power")
+        self.hp = knight_information.get("hp")
+        self.armour = knight_information.get("armour")
+        self.weapon = knight_information.get("weapon")
+        self.potion = knight_information.get("potion")
+        self.protection = 0
+
+    def apply_armour(self) -> None:
+        for value1 in self.armour:
+            for key, value in value1.items():
+                if key == "protection":
+                    self.protection += value
+
+        self.knight_information.update({"protection": self.protection})
+
+    def apply_weapon(self) -> None:
+        for key, value in self.weapon.items():
+            if key == "power":
+                self.power += value
```

Comment:
The `apply_weapon` method assumes that the weapon dictionary will always have a 'power' key. Ensure that this assumption is valid or add error handling for cases where 'power' might not be present.